test/kmeans.jl

```diff
@@ -114,4 +114,13 @@ end
     @test equal_kmresults(r, r_t)
 end
 
+x_int = rand(Int16, m, n)
```

Comment:
I'd rename it to `x` and put into "Integer data" testset. Also, we can use `@testset "$T data" for (T, TC, D, WC) in ((Int, Float64, Float64, Int), ...)` to test for more type combinations.

Actually, I don't see that `kmeans!()` is tested anywhere. Could you please also add some tests for it? E.g. would `Float32` centers work with `Float64` data? with `Int` data? Would `Int` centers fail? This was the major motivation of your PRs, so we need to test that.

Reply:
Added tests for combinations, some produce warning of objective function increasing but it's irrelevant because it's random data. 

PS: yes `Int` centers fail as per previous comments from @nalimilan there was a revert to force the centers to be of type `AbstractFloat` as otherwise it required more type gymnastic that was likely desirable. Float32 centers with Float64 data work.